File: src/services/EmailService.ts
import nodemailer from 'nodemailer';

export interface EmailConfig {
  host: string;
  port: number;
  secure: boolean;
  user: string;
  password: string;
  fromEmail: string;
  fromName: string;
}

export class EmailService {
  private transporter: nodemailer.Transporter;
  private config: EmailConfig;

  constructor() {
    this.config = {
      host: process.env.EMAIL_HOST || 'smtp.gmail.com',
      port: parseInt(process.env.EMAIL_PORT || '587'),
      secure: process.env.EMAIL_SECURE === 'true',
      user: process.env.EMAIL_HOST_USER || '',
      password: process.env.EMAIL_HOST_PASSWORD || '',
      fromEmail: process.env.DEFAULT_FROM_EMAIL || 'noreply@notarize.com',
      fromName: process.env.DEFAULT_FROM_NAME || 'RON API Service'
    };

    this.validateConfig();
    this.createTransporter();
  }

  private validateConfig(): void {
    if (!this.config.user || !this.config.password) {
      throw new Error('Email configuration is incomplete. Please set EMAIL_HOST_USER and EMAIL_HOST_PASSWORD');
    }
  }

  private createTransporter(): void {
    this.transporter = nodemailer.createTransport({
      host: this.config.host,
      port: this.config.port,
      secure: this.config.secure,
      auth: {
        user: this.config.user,
        pass: this.config.password
      },
      tls: {
        rejectUnauthorized: false
      }
    });
  }

  async verifyConnection(): Promise<boolean> {
    try {
      await this.transporter.verify();
      console.log('Email service connection verified successfully');
      return true;
    } catch (error) {
      console.error('Email service connection failed:', error);
      return false;
    }
  }

  private generateOTPEmailTemplate(otp: string, firstName: string, type: 'registration' | 'login'): string {
    const purpose = type === 'registration' ? 'complete your registration' : 'verify your login';
    const title = type === 'registration' ? 'Registration Verification' : 'Login Verification';
    
    return `
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>${title}</title>
    <style>
        body { font-family: Arial, sans-serif; line-height: 1.6; color: #333; }
        .container { max-width: 600px; margin: 0 auto; padding: 20px; }
        .header { background: #2c3e50; color: white; padding: 20px; text-align: center; }
        .content { padding: 30px; background: #f9f9f9; }
        .otp-code { 
            background: #3498db; 
            color: white; 
            padding: 15px 30px; 
            font-size: 32px; 
            font-weight: bold; 
            text-align: center; 
            margin: 20px 0; 
            border-radius: 5px;
            letter-spacing: 5px;
        }
        .warning { color: #e74c3c; font-size: 14px; margin-top: 20px; }
        .footer { text-align: center; padding: 20px; color: #666; font-size: 14px; }
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <h1>Business Intelligence Assistant</h1>
            <h2>${title}</h2>
        </div>
        <div class="content">
            <p>Hello ${firstName},</p>
            <p>Use this verification code to ${purpose}:</p>
            <div class="otp-code">${otp}</div>
            <p>This code will expire in 10 minutes for security purposes.</p>
            <div class="warning">
                <strong>Security Notice:</strong> If you didn't request this verification code, please ignore this email. 
                Never share this code with anyone.
            </div>
        </div>
        <div class="footer">
            <p>This email was sent by ${this.config.fromName}</p>
            <p>If you have questions, please contact our support team.</p>
        </div>
    </div>
</body>
</html>`;
  }

  private generateWelcomeEmailTemplate(firstName: string, accountType: string, displayName: string): string {
    return `
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Welcome to Business Intelligence Assistant</title>
    <style>
        body { font-family: Arial, sans-serif; line-height: 1.6; color: #333; }
        .container { max-width: 600px; margin: 0 auto; padding: 20px; }
        .header { background: #27ae60; color: white; padding: 20px; text-align: center; }
        .content { padding: 30px; background: #f9f9f9; }
        .features { background: white; padding: 20px; margin: 20px 0; border-radius: 5px; }
        .feature-item { margin: 15px 0; padding-left: 20px; }
        .cta-button { 
            background: #3498db; 
            color: white; 
            padding: 15px 30px; 
            text-decoration: none; 
            border-radius: 5px; 
            display: inline-block; 
            margin: 20px 0; 
        }
        .footer { text-align: center; padding: 20px; color: #666; font-size: 14px; }
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <h1>Welcome to Business Intelligence Assistant!</h1>
        </div>
        <div class="content">
            <p>Hello ${firstName},</p>
            <p>Congratulations! Your ${accountType} account has been successfully verified and activated.</p>
            <p><strong>Account Details:</strong></p>
            <ul>
                <li>Display Name: ${displayName}</li>
                <li>Account Type: ${accountType.charAt(0).toUpperCase() + accountType.slice(1)}</li>
                <li>Email: [User's email will be shown here]</li>
            </ul>
            
            <div class="features">
                <h3>What you can do now:</h3>
                <div class="feature-item">📊 Upload CSV and Excel files for analysis</div>
                <div class="feature-item">🤖 Ask questions about your data in natural language</div>
                <div class="feature-item">📈 Generate interactive charts and visualizations</div>
                <div class="feature-item">💡 Get AI-powered business insights and recommendations</div>
                <div class="feature-item">📱 Access your dashboard from any device</div>
                <div class="feature-item">🔒 Secure two-factor authentication protection</div>
            </div>
            
            <p>Ready to start analyzing your business data?</p>
            <a href="#" class="cta-button">Access Your Dashboard</a>
            
            <div style="margin-top: 30px; padding: 15px; background: #ecf0f1; border-radius: 5px;">
                <h4>Security Features:</h4>
                <p>✅ Two-factor authentication enabled<br>
                ✅ Trusted device management<br>
                ✅ Secure data encryption<br>
                ✅ Regular security monitoring</p>
            </div>
        </div>
        <div class="footer">
            <p>Thank you for choosing ${this.config.fromName}</p>
            <p>Need help? Contact our support team anytime.</p>
        </div>
    </div>
</body>
</html>`;
  }

  private generateSecurityAlertTemplate(firstName: string, alertType: string, details: string): string {
    return `
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Security Alert - Business Intelligence Assistant</title>
    <style>
        body { font-family: Arial, sans-serif; line-height: 1.6; color: #333; }
        .container { max-width: 600px; margin: 0 auto; padding: 20px; }
        .header { background: #e74c3c; color: white; padding: 20px; text-align: center; }
        .content { padding: 30px; background: #f9f9f9; }
        .alert-box { background: #fff3cd; border: 1px solid #ffeaa7; padding: 15px; margin: 20px 0; border-radius: 5px; }
        .footer { text-align: center; padding: 20px; color: #666; font-size: 14px; }
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <h1>🔒 Security Alert</h1>
        </div>
        <div class="content">
            <p>Hello ${firstName},</p>
            <p>We detected ${alertType} on your Business Intelligence Assistant account.</p>
            
            <div class="alert-box">
                <strong>Details:</strong><br>
                ${details}
            </div>
            
            <p><strong>What should you do?</strong></p>
            <ul>
                <li>If this was you, no action is needed</li>
                <li>If this wasn't you, please secure your account immediately</li>
                <li>Consider changing your password</li>
                <li>Review your trusted devices</li>
            </ul>
            
            <p>If you have any concerns, please contact our security team immediately.</p>
        </div>
        <div class="footer">
            <p>This alert was sent for your account security</p>
            <p>${this.config.fromName} Security Team</p>
        </div>
    </div>
</body>
</html>`;
  }

  async sendOTPEmail(
    email: string, 
    otp: string, 
    firstName: string, 
    type: 'registration' | 'login'
  ): Promise<boolean> {
    try {
      const subject = type === 'registration' 
        ? 'Verify Your Registration - Business Intelligence Assistant' 
        : 'Login Verification Code - Business Intelligence Assistant';

      const htmlContent = this.generateOTPEmailTemplate(otp, firstName, type);

      await this.transporter.sendMail({
        from: `"${this.config.fromName}" <${this.config.fromEmail}>`,
        to: email,
        subject,
        html: htmlContent,
        text: `Your verification code is: ${otp}. This code will expire in 10 minutes.`
      });

      console.log(`${type} OTP email sent successfully to ${email}`);
      return true;
    } catch (error) {
      console.error(`Failed to send ${type} OTP email to ${email}:`, error);
      return false;
    }
  }

  async sendWelcomeEmail(
    email: string, 
    firstName: string, 
    accountType: string, 
    displayName: string
  ): Promise<boolean> {
    try {
      const htmlContent = this.generateWelcomeEmailTemplate(firstName, accountType, displayName);

      await this.transporter.sendMail({
        from: `"${this.config.fromName}" <${this.config.fromEmail}>`,
        to: email,
        subject: 'Welcome to Business Intelligence Assistant - Account Activated!',
        html: htmlContent,
        text: `Welcome ${firstName}! Your ${accountType} account has been successfully activated. You can now start analyzing your business data with our AI-powered assistant.`
      });

      console.log(`Welcome email sent successfully to ${email}`);
      return true;
    } catch (error) {
      console.error(`Failed to send welcome email to ${email}:`, error);
      return false;
    }
  }

  async sendSecurityAlert(
    email: string, 
    firstName: string, 
    alertType: string, 
    details: string
  ): Promise<boolean> {
    try {
      const htmlContent = this.generateSecurityAlertTemplate(firstName, alertType, details);

      await this.transporter.sendMail({
        from: `"${this.config.fromName} Security" <${this.config.fromEmail}>`,
        to: email,
        subject: `🔒 Security Alert - ${alertType}`,
        html: htmlContent,
        text: `Security Alert: ${alertType}. Details: ${details}. If this wasn't you, please secure your account immediately.`
      });

      console.log(`Security alert email sent successfully to ${email}`);
      return true;
    } catch (error) {
      console.error(`Failed to send security alert email to ${email}:`, error);
      return false;
    }
  }

  async sendTestEmail(email: string): Promise<boolean> {
    try {
      await this.transporter.sendMail({
        from: `"${this.config.fromName}" <${this.config.fromEmail}>`,
        to: email,
        subject: 'Email Service Test - Business Intelligence Assistant',
        html: '<h1>Email Service Test</h1><p>If you receive this email, the email service is working correctly.</p>',
        text: 'Email Service Test - If you receive this email, the email service is working correctly.'
      });

      console.log(`Test email sent successfully to ${email}`);
      return true;
    } catch (error) {
      console.error(`Failed to send test email to ${email}:`, error);
      return false;
    }
  }
}

// Export singleton instance
export const emailService = new EmailService();

File: backend/app/core/email_service.py
"""
Email Service Module - Professional email sending with HTML templates
"""

import aiosmtplib
import logging
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
from email.mime.base import MIMEBase
from email import encoders
from typing import List, Optional, Dict, Any
from jinja2 import Template
from datetime import datetime

from .config import settings

logger = logging.getLogger(__name__)


class EmailService:
    """
    Professional email service with HTML templates and async SMTP
    """
    
    def __init__(self):
        self.smtp_host = settings.SMTP_HOST
        self.smtp_port = settings.SMTP_PORT
        self.smtp_user = settings.SMTP_USER
        self.smtp_password = settings.SMTP_PASSWORD
        self.smtp_tls = settings.SMTP_TLS
        self.email_from = settings.EMAIL_FROM
        self.email_from_name = settings.EMAIL_FROM_NAME
    
    async def send_email(
        self,
        to_email: str,
        subject: str,
        html_content: str,
        text_content: Optional[str] = None,
        attachments: Optional[List[Dict[str, Any]]] = None
    ) -> bool:
        """
        Send email with HTML content and optional attachments
        """
        try:
            # Create message
            message = MIMEMultipart("alternative")
            message["Subject"] = subject
            message["From"] = f"{self.email_from_name} <{self.email_from}>"
            message["To"] = to_email
            
            # Add text part (fallback)
            if text_content:
                text_part = MIMEText(text_content, "plain")
                message.attach(text_part)
            
            # Add HTML part
            html_part = MIMEText(html_content, "html")
            message.attach(html_part)
            
            # Add attachments if any
            if attachments:
                for attachment in attachments:
                    self._add_attachment(message, attachment)
            
            # Send email
            await self._send_message(message, to_email)
            
            logger.info(f"Email sent successfully to {to_email}")
            return True
            
        except Exception as e:
            logger.error(f"Failed to send email to {to_email}: {str(e)}")
            return False
    
    async def send_registration_otp(self, email: str, name: str, otp_code: str) -> bool:
        """
        Send registration OTP verification email
        """
        subject = "Welcome to INSIGHTORA - Verify Your Account"
        
        html_content = self._get_registration_otp_template(name, otp_code)
        text_content = f"""
Welcome to INSIGHTORA, {name}!

Your verification code is: {otp_code}

Please enter this code to complete your registration.
This code will expire in 10 minutes.

If you didn't create this account, please ignore this email.

Best regards,
INSIGHTORA Team
        """.strip()
        
        return await self.send_email(email, subject, html_content, text_content)
    
    async def send_login_otp(self, email: str, name: str, otp_code: str) -> bool:
        """
        Send login OTP verification email
        """
        subject = "INSIGHTORA - Login Verification Code"
        
        html_content = self._get_login_otp_template(name, otp_code)
        text_content = f"""
Hello {name},

Your login verification code is: {otp_code}

Please enter this code to complete your login.
This code will expire in 10 minutes.

If you didn't attempt to log in, please secure your account immediately.

Best regards,
INSIGHTORA Team
        """.strip()
        
        return await self.send_email(email, subject, html_content, text_content)
    
    async def send_welcome_email(self, email: str, name: str) -> bool:
        """
        Send welcome email after successful registration
        """
        subject = "Welcome to INSIGHTORA - Your BI Journey Starts Here!"
        
        html_content = self._get_welcome_template(name)
        text_content = f"""
Welcome to INSIGHTORA, {name}!

Your account has been successfully verified and you're now ready to explore the power of AI-driven Business Intelligence.

What you can do with INSIGHTORA:
• Connect your databases and data sources
• Ask questions in natural language
• Get instant insights and visualizations
• Create interactive dashboards
• Share reports with your team

Get started by visiting your dashboard and connecting your first data source.

Need help? Our documentation and support team are here to assist you.

Best regards,
The INSIGHTORA Team
        """.strip()
        
        return await self.send_email(email, subject, html_content, text_content)
    
    def _get_registration_otp_template(self, name: str, otp_code: str) -> str:
        """
        Professional registration OTP email template
        """
        template = Template("""
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Verify Your INSIGHTORA Account</title>
    <style>
        * { margin: 0; padding: 0; box-sizing: border-box; }
        body { font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif; line-height: 1.6; color: #333; }
        .container { max-width: 600px; margin: 0 auto; padding: 20px; }
        .header { background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); color: white; padding: 30px; text-align: center; border-radius: 10px 10px 0 0; }
        .logo { font-size: 28px; font-weight: bold; margin-bottom: 10px; }
        .content { background: white; padding: 40px 30px; border-radius: 0 0 10px 10px; box-shadow: 0 4px 6px rgba(0,0,0,0.1); }
        .otp-box { background: #f8f9ff; border: 2px dashed #667eea; padding: 20px; margin: 25px 0; text-align: center; border-radius: 8px; }
        .otp-code { font-size: 32px; font-weight: bold; color: #667eea; letter-spacing: 8px; margin: 10px 0; }
        .footer { text-align: center; margin-top: 30px; color: #666; font-size: 14px; }
        .button { display: inline-block; background: #667eea; color: white; padding: 12px 24px; text-decoration: none; border-radius: 6px; margin: 20px 0; }
        .warning { background: #fff3cd; border: 1px solid #ffeaa7; padding: 15px; border-radius: 6px; margin: 20px 0; color: #856404; }
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <div class="logo">🔍 INSIGHTORA</div>
            <p>Business Intelligence Made Simple</p>
        </div>
        
        <div class="content">
            <h2 style="color: #333; margin-bottom: 20px;">Welcome, {{ name }}! 🎉</h2>
            
            <p>Thank you for signing up for INSIGHTORA! To complete your registration and start exploring AI-powered business intelligence, please verify your email address.</p>
            
            <div class="otp-box">
                <p style="margin-bottom: 10px; font-weight: bold;">Your Verification Code:</p>
                <div class="otp-code">{{ otp_code }}</div>
                <p style="margin-top: 10px; font-size: 14px; color: #666;">This code expires in 10 minutes</p>
            </div>
            
            <p>Simply enter this code in the verification form to activate your account and start your journey with intelligent data analysis.</p>
            
            <div class="warning">
                <strong>Security Notice:</strong> If you didn't create this account, please ignore this email. Your security is our priority.
            </div>
            
            <h3 style="color: #667eea; margin: 25px 0 15px 0;">What's Next?</h3>
            <ul style="padding-left: 20px; margin-bottom: 20px;">
                <li>✅ Verify your email with the code above</li>
                <li>🔗 Connect your data sources</li>
                <li>💬 Ask questions in natural language</li>
                <li>📊 Get instant visualizations</li>
                <li>🚀 Build powerful dashboards</li>
            </ul>
        </div>
        
        <div class="footer">
            <p>© {{ current_year }} INSIGHTORA. All rights reserved.</p>
            <p>Making data insights accessible to everyone</p>
        </div>
    </div>
</body>
</html>
        """.strip())
        
        return template.render(
            name=name,
            otp_code=otp_code,
            current_year=datetime.now().year
        )
    
    def _get_login_otp_template(self, name: str, otp_code: str) -> str:
        """
        Professional login OTP email template
        """
        template = Template("""
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>INSIGHTORA Login Verification</title>
    <style>
        * { margin: 0; padding: 0; box-sizing: border-box; }
        body { font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif; line-height: 1.6; color: #333; }
        .container { max-width: 600px; margin: 0 auto; padding: 20px; }
        .header { background: linear-gradient(135deg, #4facfe 0%, #00f2fe 100%); color: white; padding: 25px; text-align: center; border-radius: 10px 10px 0 0; }
        .logo { font-size: 28px; font-weight: bold; margin-bottom: 10px; }
        .content { background: white; padding: 35px 30px; border-radius: 0 0 10px 10px; box-shadow: 0 4px 6px rgba(0,0,0,0.1); }
        .otp-box { background: #f0f9ff; border: 2px solid #4facfe; padding: 20px; margin: 25px 0; text-align: center; border-radius: 8px; }
        .otp-code { font-size: 32px; font-weight: bold; color: #4facfe; letter-spacing: 8px; margin: 10px 0; }
        .footer { text-align: center; margin-top: 30px; color: #666; font-size: 14px; }
        .security-notice { background: #fee2e2; border: 1px solid #fecaca; padding: 15px; border-radius: 6px; margin: 20px 0; color: #991b1b; }
        .info-box { background: #f0f9ff; border-left: 4px solid #4facfe; padding: 15px; margin: 20px 0; }
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <div class="logo">🔍 INSIGHTORA</div>
            <p>Secure Login Verification</p>
        </div>
        
        <div class="content">
            <h2 style="color: #333; margin-bottom: 20px;">Login Verification Required</h2>
            
            <p>Hello <strong>{{ name }}</strong>,</p>
            <p>Someone is attempting to log into your INSIGHTORA account. For your security, please enter the verification code below:</p>
            
            <div class="otp-box">
                <p style="margin-bottom: 10px; font-weight: bold;">Your Login Code:</p>
                <div class="otp-code">{{ otp_code }}</div>
                <p style="margin-top: 10px; font-size: 14px; color: #666;">Valid for 10 minutes only</p>
            </div>
            
            <div class="info-box">
                <p><strong>Login Details:</strong></p>
                <p>🕐 Time: {{ current_time }}</p>
                <p>📅 Date: {{ current_date }}</p>
            </div>
            
            <div class="security-notice">
                <strong>⚠️ Security Alert:</strong> If you didn't attempt to log in, please contact our support team immediately and consider changing your password.
            </div>
            
            <p>After entering this code, you'll have full access to your INSIGHTORA dashboard and all your business intelligence tools.</p>
        </div>
        
        <div class="footer">
            <p>© {{ current_year }} INSIGHTORA. All rights reserved.</p>
            <p>Need help? Contact our support team</p>
        </div>
    </div>
</body>
</html>
        """.strip())
        
        now = datetime.now()
        return template.render(
            name=name,
            otp_code=otp_code,
            current_time=now.strftime("%I:%M %p"),
            current_date=now.strftime("%B %d, %Y"),
            current_year=now.year
        )
    
    def _get_welcome_template(self, name: str) -> str:
        """
        Professional welcome email template
        """
        template = Template("""
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Welcome to INSIGHTORA</title>
    <style>
        * { margin: 0; padding: 0; box-sizing: border-box; }
        body { font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif; line-height: 1.6; color: #333; }
        .container { max-width: 600px; margin: 0 auto; padding: 20px; }
        .header { background: linear-gradient(135deg, #a8edea 0%, #fed6e3 100%); color: #333; padding: 40px; text-align: center; border-radius: 10px 10px 0 0; }
        .logo { font-size: 32px; font-weight: bold; margin-bottom: 15px; }
        .content { background: white; padding: 40px 30px; border-radius: 0 0 10px 10px; box-shadow: 0 4px 6px rgba(0,0,0,0.1); }
        .feature-grid { display: grid; grid-template-columns: 1fr 1fr; gap: 20px; margin: 30px 0; }
        .feature-box { background: #f8f9fa; padding: 20px; border-radius: 8px; text-align: center; }
        .feature-icon { font-size: 32px; margin-bottom: 10px; }
        .cta-button { display: inline-block; background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); color: white; padding: 15px 30px; text-decoration: none; border-radius: 8px; margin: 25px 0; font-weight: bold; }
        .footer { text-align: center; margin-top: 30px; color: #666; font-size: 14px; }
        .tips { background: #fff9e6; border-left: 4px solid #ffd700; padding: 20px; margin: 25px 0; }
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <div class="logo">🎉 Welcome to INSIGHTORA!</div>
            <p style="font-size: 18px; margin-top: 10px;">Your AI-Powered Business Intelligence Platform</p>
        </div>
        
        <div class="content">
            <h2 style="color: #333; margin-bottom: 20px;">Hello {{ name }}, Welcome Aboard! 🚀</h2>
            
            <p>Congratulations! Your INSIGHTORA account is now active and ready to transform how you interact with your business data.</p>
            
            <div class="feature-grid">
                <div class="feature-box">
                    <div class="feature-icon">🔗</div>
                    <h4>Connect Data</h4>
                    <p>Link databases, upload files, and integrate APIs</p>
                </div>
                <div class="feature-box">
                    <div class="feature-icon">💬</div>
                    <h4>Ask Questions</h4>
                    <p>Use natural language to query your data</p>
                </div>
                <div class="feature-box">
                    <div class="feature-icon">📊</div>
                    <h4>Get Insights</h4>
                    <p>Receive instant visualizations and analytics</p>
                </div>
                <div class="feature-box">
                    <div class="feature-icon">🎯</div>
                    <h4>Build Dashboards</h4>
                    <p>Create interactive reports and share with your team</p>
                </div>
            </div>
            
            <div class="tips">
                <h3 style="color: #b8860b; margin-bottom: 15px;">💡 Quick Start Tips:</h3>
                <ul style="padding-left: 20px;">
                    <li><strong>Start Small:</strong> Upload a CSV file to get familiar with the platform</li>
                    <li><strong>Ask Simple Questions:</strong> Try "Show me sales by month" or "What are my top products?"</li>
                    <li><strong>Explore Templates:</strong> Use our pre-built dashboard templates</li>
                    <li><strong>Join the Community:</strong> Connect with other users for tips and best practices</li>
                </ul>
            </div>
            
            <div style="text-align: center;">
                <a href="#" class="cta-button">Get Started with Your First Dashboard 🎯</a>
            </div>
            
            <h3 style="color: #667eea; margin: 30px 0 15px 0;">What Makes INSIGHTORA Special?</h3>
            <ul style="padding-left: 20px; margin-bottom: 25px;">
                <li>🤖 <strong>AI-Powered:</strong> Natural language queries with advanced LLM technology</li>
                <li>⚡ <strong>Real-Time:</strong> Live data updates and instant visualizations</li>
                <li>🔒 <strong>Secure:</strong> Enterprise-grade security with role-based access</li>
                <li>📱 <strong>Mobile-Ready:</strong> Access your dashboards anywhere, anytime</li>
                <li>🔧 <strong>Customizable:</strong> Build exactly what your business needs</li>
            </ul>
            
            <p>We're excited to see what insights you'll discover! If you have any questions, our support team is here to help.</p>
        </div>
        
        <div class="footer">
            <p>© {{ current_year }} INSIGHTORA. All rights reserved.</p>
            <p>Making data insights accessible to everyone 📈</p>
            <p style="margin-top: 10px;">
                <a href="#" style="color: #667eea; text-decoration: none;">Documentation</a> | 
                <a href="#" style="color: #667eea; text-decoration: none;">Support</a> | 
                <a href="#" style="color: #667eea; text-decoration: none;">Community</a>
            </p>
        </div>
    </div>
</body>
</html>
        """.strip())
        
        return template.render(
            name=name,
            current_year=datetime.now().year
        )
    
    async def _send_message(self, message: MIMEMultipart, to_email: str):
        """
        Send email message via SMTP
        """
        try:
            # Create SMTP connection with correct TLS settings for Gmail
            smtp = aiosmtplib.SMTP(
                hostname=self.smtp_host,
                port=self.smtp_port,
                start_tls=True,  # Use STARTTLS instead of use_tls
                validate_certs=False  # For development, can be True in production
            )
            
            # Connect and authenticate
            await smtp.connect()
            if self.smtp_user and self.smtp_password:
                await smtp.login(self.smtp_user, self.smtp_password)
            
            # Send message
            await smtp.send_message(message)
            
            # Close connection
            await smtp.quit()
            
        except Exception as e:
            logger.error(f"SMTP error sending to {to_email}: {str(e)}")
            raise
    
    def _add_attachment(self, message: MIMEMultipart, attachment: Dict[str, Any]):
        """
        Add attachment to email message
        """
        try:
            with open(attachment["path"], "rb") as file:
                part = MIMEBase("application", "octet-stream")
                part.set_payload(file.read())
                encoders.encode_base64(part)
                part.add_header(
                    "Content-Disposition",
                    f"attachment; filename= {attachment['filename']}"
                )
                message.attach(part)
        except Exception as e:
            logger.error(f"Failed to add attachment {attachment.get('filename', 'unknown')}: {str(e)}")


# Global email service instance
email_service = EmailService()